client/elements/text/sc-text-image.html

```diff
@@ -62,11 +62,19 @@
                   },
                   viewer: {
                       type: Object
+                  },
+                  lastDetail: {
+                      type: Object
                   }
               }
           }
 
           showImage(detail) {
+            if (!detail || (this.lastDetail && detail.vol === this.lastDetail.vol 
```

Comment:
similarly, I'd extract the second part of this complex if statement to a separate function (named `_didDetailChange` for example) to make it easier to read.